pkg/audit/manager.go

```diff
@@ -187,32 +185,25 @@ func New(mgr manager.Manager, deps *Dependencies) (*Manager, error) {
 		expansionSystem: deps.ExpansionSystem,
 		pubsubSystem:    deps.PubSubSystem,
 	}
-	return am, nil
+	return am, am.reporter.registerCallback()
 }
 
 // audit performs an audit then updates the status of all constraint resources with the results.
 func (am *Manager) audit(ctx context.Context) error {
-	startTime := time.Now()
-	timestamp := startTime.UTC().Format(time.RFC3339)
+	am.reporter.startTime = time.Now()
+	timestamp := am.reporter.startTime.UTC().Format(time.RFC3339)
 	am.log = log.WithValues(logging.AuditID, timestamp)
 	logStart(am.log)
 	// record audit latency
 	defer func() {
 		logFinish(am.log)
-		endTime := time.Now()
-		latency := endTime.Sub(startTime)
+		am.reporter.endTime = time.Now()
+		latency := am.reporter.endTime.Sub(am.reporter.startTime)
 		if err := am.reporter.reportLatency(latency); err != nil {
 			am.log.Error(err, "failed to report latency")
 		}
-		if err := am.reporter.reportRunEnd(endTime); err != nil {
```

Comment:
We should keep the function calls as-is as opposed to setting variables directly.

In general, if there has been any change to any non-reporter file (particularly if its a file adjacent to a `*_reporter.go` file), that is likely something to be avoided.

In this case, we do not want to be writing directly to the reporter objects state, since that eliminates the ability of the reporter to abstract away issues like concurrency (which is a concern in this case b/c callbacks are likely called from a different goroutine).